src/predicates/intersects.jl

```diff
@@ -106,22 +107,94 @@ function intersects(g₁::Geometry{Dim,T}, g₂::Geometry{Dim,T}) where {Dim,T}
       AB = B - A
       AO = O - A
       d = AB × AO × AB
-    else # simplex case
-      C, B, A = points
-      AB = B - A
-      AC = C - A
-      AO = O - A
-      ABᵀ = AC × AB × AB
-      ACᵀ = AB × AC × AC
-      if ABᵀ ⋅ AO > zero(T)
-        popat!(points, 1) # pop C
-        d = ABᵀ
-      elseif ACᵀ ⋅ AO > zero(T)
-        popat!(points, 2) # pop B
-        d = ACᵀ
-      else
-        return true
-      end
+    else
+      d = gjk!(Val(Dim), O, points)
+      isnothing(d) && return true
+    end
+  end
+end
+
+"""
+    gjk!(::Val{Dim}, origin, points) where Dim
+
+Perform one iteration of the GJK algorithm.
+
+It returns `nothing` if the `Dim`-simplex represented by `points`
+contains the origin point. Otherwise, it returns a vector with
+the direction for searching the next point.
+
+If the simplex is complete, it removes one point from the set to
+make room for the next point. A complete simplex must have `Dim + 1` points. 
+
+See also [`intersects`](@ref).
+"""
+function gjk! end
+
+function gjk!(::Val{2}, origin, points)
```

Comment:
Yes. That's what I want to do. However, I'll need to modify the functions `minkowskipoint` and `minkowskiorigin` because they make the 2D case dispatch on the `origin::Point{3,T}`.